app/components/slack-context-menu.js

```diff
@@ -0,0 +1,165 @@
+(function () {
+	'use strict';
+	// Load in our dependencies
+	var gui = window.require('nw.gui');
+	var clipboard = gui.Clipboard.get();
+
+	var GOOGLE_SEARCH_URL = "https://www.google.com/#q=";
+
+	function isSelected(target) {
+		return target.selectionStart != target.selectionEnd;
+	}
+
+	function getSelectedValue(target) {
+		return target.value.substring(target.selectionStart, target.selectionEnd);
+	}
+
+	function removeSelectedValue(target) {
+		var selectionStart = target.selectionStart;
+		var selectedValue = target.value.substring(target.selectionStart, target.selectionEnd);
+		target.value = target.value.substring(0, target.selectionStart) + target.value.substring(target.selectionEnd);
+		target.setSelectionRange(selectionStart,selectionStart); // place cursor at original selectionStart
+		return selectedValue;
+	}
+
+	function pasteValue(target, value) {
+		var selectionStart = target.selectionStart + value.length;
+		target.value = target.value.substring(0, target.selectionStart) + value + target.value.substring(target.selectionEnd);
+		target.setSelectionRange(selectionStart,selectionStart); // place cursor at original selectionStart
+	}
+
+	function buildTextareaMenu(menu, target) {
+		var selected = isSelected(target);
+
+		// Select all
+		menu.append(new gui.MenuItem({
+			label: 'Select All',
+			click: function () {
+				target.select();
+			}
+		}));
+		// TODO: Undo
+		// TODO: Redo
```

Comment:
Instead of leaving these as TODO's in the code. Can we relocate them to a GitHub issue (it will lead to better transparency)?